planet/clients/orders.py

```diff
@@ -337,71 +340,90 @@ async def download_order(
                      for location in locations]
         return filenames
 
-    async def poll(
+    async def wait(
         self,
         order_id: str,
-        state: str = None,
-        wait: int = 1,
-        report=None
+        states: typing.Iterable[str] = ORDERS_STATES_COMPLETE,
+        delay: int = 5,
+        max_attempts: int = 200,
+        report: typing.Callable[[str], None] = None
```

Comment:
How generalizing this to be named `callback`? Reporting is one kind of use but not necessarily the only one. And maybe it could be passed the entire order description and not only the status?

Reply:
Was reminded by @jreiberkyle that an order's state is the only changing attribute, so there's no real need for the entire description in the callback.

Reply:
yes, changing it to `callback` is a great idea